File: work/deprecated/room/converter/WeatherCodeConverter.java
package wangdaye.com.geometricweather.room.converter;

import androidx.room.TypeConverter;

import wangdaye.com.geometricweather.basic.models.weather.WeatherCode;

public class WeatherCodeConverter {

    @TypeConverter
    public static WeatherCode convertToEntityProperty(String databaseValue) {
        return WeatherCode.valueOf(databaseValue);
    }

    @TypeConverter
    public static String convertToDatabaseValue(WeatherCode entityProperty) {
        return entityProperty.name();
    }
}

File: app/src/main/java/org/breezyweather/db/converters/WeatherCodeConverter.kt
/**
 * This file is part of Breezy Weather.
 *
 * Breezy Weather is free software: you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by the
 * Free Software Foundation, either version 3 of the License, or (at your
 * option) any later version.
 *
 * Breezy Weather is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
 * License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
 */

package org.breezyweather.db.converters

import io.objectbox.converter.PropertyConverter
import org.breezyweather.common.basic.models.weather.WeatherCode

class WeatherCodeConverter : PropertyConverter<WeatherCode?, String?> {
    override fun convertToEntityProperty(databaseValue: String?): WeatherCode? =
        if (databaseValue.isNullOrEmpty()) null else WeatherCode.getInstance(databaseValue)

    override fun convertToDatabaseValue(entityProperty: WeatherCode?): String? =
        entityProperty?.id
}
